tests/TypeParseTest.php

```diff
@@ -216,6 +216,23 @@ public function testIntersectionOfIterables()
         $this->assertSame('iterable<mixed, A>&iterable<mixed, B>', (string) Type::parseString('iterable<A>&iterable<B>'));
     }
 
+    /**
+     * @return void
+     */
+    public function testIntersectionOfObjectLike()
+    {
+        $this->assertSame('array{a: int, b: int}', (string) Type::parseString('array{a: int}&array{b: int}'));
```

Comment:
hmm I just realized that there is not enough test coverage. Let me add a few more cases.

Reply:
all right, i did find a couple of issues ^^;; it should be better now.